### PR TITLE
fix: partial revert #1058

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

#1058 it's not working for me, I tried to add a new subcommand to `internal/cli/config/config.go` and I keep getting 

```bash
==> Linting changed go files...
internal/cli/config/config.go:159:3: undeclared name: `SetBuilder` (typecheck)
		SetBuilder(),
		^
internal/cli/config/config.go:160:3: undeclared name: `ListBuilder` (typecheck)
		ListBuilder(),
		^
internal/cli/config/config.go:161:3: undeclared name: `DescribeBuilder` (typecheck)
		DescribeBuilder(),
		^
internal/cli/config/config.go:162:3: undeclared name: `RenameBuilder` (typecheck)
		RenameBuilder(),
		^
internal/cli/config/config.go:163:3: undeclared name: `DeleteBuilder` (typecheck)
		DeleteBuilder(),
		^
internal/cli/config/config.go:164:3: undeclared name: `EditBuilder` (typecheck)
		EditBuilder(),
		^
internal/cli/config/config_test.go:30:3: undeclared name: `Builder` (typecheck)
		Builder(),
		^
internal/mongosh/mongosh_bin_windows.go:17:7: `mongoshBin` is unused (deadcode)
const mongoshBin = "mongosh.exe"
      ^
==> Done...

```

It seems like it fails to see the context of the changes and failing to incorporate those into the lint 

Also removing the per file check uncovered some missing lint failures

### CODE REVIEW

File: cmd/atlas/atlas.go [hunk 1]
```diff
@@ -62,25 +63,25 @@ func createConfigFromMongoCLIConfig() {
 		return
 	}
 
-	atlasConfigPath := fmt.Sprintf("%s/%s", atlasConfigHomePath, "config.toml")
+	atlasConfigPath := path.Join(atlasConfigHomePath, "config.toml")
```
Comment (gssbzn) [inline cmd/atlas/atlas.go:66]: drive-by

File: cmd/atlas/atlas.go [hunk 2]
```diff
@@ -62,25 +63,25 @@ func createConfigFromMongoCLIConfig() {
...
 
-	in, err := os.Open(path)
+	in, err := os.Open(p)
 	if err != nil {
 		return
 	}
 	defer in.Close()
 
-	_, _ = fmt.Fprintf(os.Stderr, `AtlasCLI has found an existing MongoCLI configuration file, copying its content to: %s
+	_, _ = fmt.Fprintf(os.Stderr, `Atlas CLI has found an existing MongoDB CLI configuration file, copying its content to: %s
```
Comment (gssbzn) [inline cmd/atlas/atlas.go:84]: copy fix drive by

File: cmd/atlas/atlas.go [hunk 3]
```diff
@@ -108,15 +109,15 @@ func createConfigFromMongoCLIConfig() {
...
+		configPath = path.Join(configDir, "config.toml")
 	}
 
 	// Check if file exists, if any error is detected try to get older file
 	if _, err := os.Stat(configPath); err == nil {
 		return configPath, nil
 	}
 
-	if configDir, err := config.OldMongoCLIConfigHome(); err == nil {
+	if configDir, err := config.OldMongoCLIConfigHome(); err == nil { //nolint:staticcheck // check if old config exists
```
Comment (gssbzn) [inline cmd/atlas/atlas.go:120]: had to add this when I enabled global linting again, the flagging the method as deprecated means you should not use it

File: cmd/mongocli/mongocli.go [hunk 4]
```diff
@@ -46,19 +47,19 @@ func updateMongoCLIConfigPath() {
 		return
 	}
 
-	mongoCLIConfigPath := fmt.Sprintf("%s/%s", mongoCLIConfigHome, "config.toml")
+	mongoCLIConfigPath := path.Join(mongoCLIConfigHome, "config.toml")
```
Comment (gssbzn) [inline cmd/mongocli/mongocli.go:50]: same changes as for atlas

**Review by andreaangiolillo — APPROVED** [review]
LGTM
